### PR TITLE
Revamp to likecoin-js

### PR DESCRIPTION
**Pull request by williamchong**
Add arweave, iscn and tx module
Maintain likepay qr code functionalities for this version, which we plan to remove

### CODE REVIEW

File: src/arweave/api.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  if (typeof files === 'string') {
+    if (!isNode()) throw new Error('file path is only supported in node.js');
+    form = await createNewForm(files);
+  } else {
+    const fileArray = Array.from(files);
+    form = await createNewForm();
+    fileArray.forEach((f) => {
+      form.append(f.name, f, f.name);
+    });
+  }
```
Comment (WeiJun0827) [inline src/arweave/api.ts:24]: Maybe could refactor this part to a new function for the use of `estimateArweavePrice()` and `uploadToArweave()`?

Comment (williamchong) [reply src/arweave/api.ts:24]: Indeed, but lets invoke rule of three and refactor later

Comment (WeiJun0827) [reply src/arweave/api.ts:24]: OK!

**Review by WeiJun0827 — APPROVED** [review]

File: src/types.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+export type ArweavePriceWithLIKE = {
+  key?: string;
+  arweaveId?: string;
+  AR: string;
+  LIKE: string;
+  list?: ArweavePriceWithLIKE[];
+}
+
+export type estimateArweaveResponse = {
```
Comment (nwingt) [inline src/types.ts:22]:
```suggestion
export type EstimateArweaveResponse = {
```

File: src/types.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  arweaveId?: string;
+  AR: string;
+  LIKE: string;
+  list?: ArweavePriceWithLIKE[];
+  ipfsHash: string;
+  memo: string;
+  address: string;
+}
+
+export type uploadArweaveResponse = {
```
Comment (nwingt) [inline src/types.ts:33]:
```suggestion
export type UploadArweaveResponse = {
```

**Review by nwingt — CHANGES_REQUESTED** [review]

**Review by nwingt — APPROVED** [review]

**Pull request merged**